src/robot/api/deco.py

```diff
@@ -41,8 +53,28 @@ def exposed_as_keyword():
 not_keyword.robot_not_keyword = True
 
 
+# Decorator used without arguments, i.e. @keyword()
+@overload
+def keyword(name: F) -> F:
+    ...
+
+
+# Decorator used with arguments, e.g. @keyword("Custom Keyword Name")
+@overload
+def keyword(
+    name: 'str | None' = None,
+    tags: Sequence[str] = (),
+    types: 'Sequence[str] | Mapping[str, Any]' = (),
+) -> Callable[[F], F]:
+    ...
+
+
 @not_keyword
-def keyword(name=None, tags=(), types=()):
+def keyword(
+    name: 'str | Callable[..., Any] | None' = None,
+    tags: Sequence[str] = (),
+    types: 'Sequence[str] | Mapping[str, Any]' = (),
+) -> 'F | Callable[[F], F]':
```

Comment:
With the above overloads, typing could be removed from this function altogether to keep it simpler.If typing is preserved, it should use type aliases and not complex constructs like `Callable[..., Any]`.

Reply:
Removing the annotations from the actual function results in a lot of typing errors; the type-checkers check if the signatures of all the defined `overload`s match up with the signature / annotations of the overloaded function.

Do you have a suggestion for the names for type aliases we could use here? It currently looks like this:
```python
def keyword(
    name: 'Callable[[F], F] | str | None' = None,
    tags: Sequence[str] = (),
    types: 'TypeHints | None' = (),
) -> 'F | Callable[[F], F]':
```
I think `Callable[[F], F]` is what we would need an alias with a clear name for.

Reply:
I guess the current code is fine. We can experiment with an alias for `Callable[[F], F]`, though.